src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -445,39 +451,79 @@ def app_events(
         help="Restrict results to a specific scope name. Can be specified multiple times.",
     ),
     first: int = typer.Option(
-        default=0, help="Fetch only the first N events. Cannot be used with --last."
+        default=-1,
+        show_default=False,
+        help="Fetch only the first N events. Cannot be used with --last.",
     ),
     last: int = typer.Option(
-        default=0, help="Fetch only the last N events. Cannot be used with --first."
+        default=-1,
+        show_default=False,
+        help="Fetch only the last N events. Cannot be used with --first.",
+    ),
+    follow: bool = typer.Option(
+        False,
+        "--follow",
+        "-f",
+        help=(
+            f"Continue polling for events. Implies --last {DEFAULT_EVENT_FOLLOW_LAST} "
+            f"unless overridden or the --since flag is used."
+        ),
+    ),
+    follow_interval: int = typer.Option(
+        10,
+        help=f"Polling interval in seconds when using the --follow flag.",
     ),
     **options,
 ):
     """Fetches events for this app from the event table configured in Snowflake."""
-    if first and last:
+    if first != -1 and last != -1:
```

Comment:
nit: can we use first >= 0 and last >= 0 here? Technically the user could decide to pass in any int, might as well treat anything negative as disabling the option.

Reply:
fixed